File: 4216-WeightedWordMapping/4216-WeightedWordMapping.java
// Last updated: 6/18/2026, 4:19:17 PM
class Solution {
    public String mapWordWeights(String[] words, int[] weights) {
        String result = "";
        for(String st : words){
            int sum = 0;
            for(int i=0; i<st.length(); i++){
                char ch = st.charAt(i);
                sum += weights[ch - 'a'];
            }
            result +=(char)('z' - (sum%26));
        }
        return result;
    }
}

File: 4216-WeightedWordMapping/4216-WeightedWordMapping.cpp
// Last updated: 6/18/2026, 4:19:20 PM
class Solution {
public:
    string mapWordWeights(vector<string>& words, vector<int>& weights) {
        string result = "";
        for(string &st : words){
            int sum = 0;
            for(char ch : st){
                sum += weights[ch - 'a'];
            }
            result += 'z' - (sum%26);
        }
        return result;
    }
};